test/timeout.test.ts

```diff
@@ -0,0 +1,111 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// The module 'assert' provides assertion methods from node
+import * as assert from 'assert';
+import { rejectOnTimeout, valueOnTimeout } from '../src/utils/timeout';
+
+suite("timeout Tests", () => {
+    suite("rejectOnTimeout", () => {
+
+        test("executes synchronously", async () => {
+            let executed: boolean;
+
+            await rejectOnTimeout(1, () => {
+                executed = true;
+            });
+
+            assert.equal(executed, true);
+        });
+
+        test("executes synchronously in promise", async () => {
+            let executed = false;
+
+            await rejectOnTimeout(1, () => {
+                return new Promise((resolve, reject) => {
+                    executed = true;
+                    resolve();
+                });
+            })
+
+            assert.equal(executed, true);
+        });
+
+        test("executes before time-out", async () => {
+            let executed = false;
+
+            await rejectOnTimeout(1, () => {
+                return new Promise((resolve, reject) => {
+                    executed = true;
+                    resolve();
+                });
+            })
+            assert.equal(executed, true);
+        });
+
+        test("timed out", async () => {
+            let executed = false;
+
+            try {
```

Comment:
Rather than doing this try/catch stuff, you can just use `assert.throws`, which validates that an exception is thrown when a block of code is executed

Reply:
Tried that, but it doesn't seem to handle async functions.